mobile/ios/App/App/NodeRunner.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  NodeRunner.swift
+//  App
+//
+//  Created by Juergen Pona on 06.05.2025.
+//
+
+import Foundation
+import NodeMobile
+
+@objc public class NodeRunner: NSObject {
+    private static var nodeStarted = false
+
+    @objc public static func startNode() {
+        // Prevent double-start
+        guard !nodeStarted else { return }
+        nodeStarted = true
+
+        // Run on a background thread so we don’t block the UI
+        DispatchQueue.global(qos: .default).async {
+                // —— DEBUG: list contents of nodejs-project folder ——
+                if let folderURL = Bundle.main.resourceURL?.appendingPathComponent("nodejs-project") {
+                    if let items = try? FileManager.default.contentsOfDirectory(atPath: folderURL.path) {
+                        print("📂 nodejs-project contains: \(items)")
+                    } else {
+                        print("❌ Could not read nodejs-project folder at \(folderURL.path)")
+                    }
+                } else {
+                    print("❌ nodejs-project folder not found in bundle")
+                }
+
+                // —— attempt to find index.mjs ——
+                guard let scriptPath = Bundle.main.path(
+                        forResource: "index",
+                        ofType: "mjs",
+                        inDirectory: "nodejs-project"
+                      ) else {
+                    print("❌ NodeRunner: could not find index.mjs in nodejs-project")
+                    return
+                }
+
+
+            // Prepare argv: ["node", "<full-path-to-index.mjs>"]
+            let args = ["node", "--no-experimental-fetch", scriptPath]
+            let argc = Int32(args.count)
+            let argv = UnsafeMutablePointer<UnsafeMutablePointer<Int8>?>.allocate(capacity: Int(argc))
+            for (i, arg) in args.enumerated() {
+                argv[i] = strdup(arg)
+            }
+
+            // Launch the embedded Node.js runtime
+            node_start(argc, argv)
```

Comment:
Can we link libnode as dynamic library into the main app executable, like we do on Android, instead of starting node as separate process here?

(Should happen in a new PR)

Reply:
A thread is still a lot better than a different process, because the lifetime of the thread is automatically tied to the main process, whereas subprocesses are normally not. But that seems to be not an issue on iOS, because iOS will kill any subprocesses together with the main app, as I'm reading.

So, we can keep this solution for now, if it works well